internal/wasm/jit/asm/amd64/impl.go

```diff
@@ -0,0 +1,382 @@
+package amd64
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/tetratelabs/wazero/internal/wasm/jit/asm"
+)
+
+// nodeImpl implements asm.Node for amd64.
+type nodeImpl struct {
+	instruction asm.Instruction
+
+	// offsetInBinary represents the offset of this node in the final binary.
+	offsetInBinary int64
```

Comment:
add comment why the offset is signed or make it unsigned? also are we expecting a binary this size? if not, say why 64bit